lib/Agrammon/DataSource/JSON.pm6

```diff
@@ -0,0 +1,54 @@
+use v6;
+use JSON::Fast;
+
+use Agrammon::Inputs;
+use Agrammon::DataSource::Util;
+
+class Agrammon::DataSource::JSON {
+    method load($simulation-name, $dataset-id, $json-data) {
+        my %input-data = from-json $json-data;
+        my $inputs = Agrammon::Inputs.new(:$simulation-name, :$dataset-id);
+        for %input-data.kv -> $full-tax, $module-data {
+            if $full-tax.index('[') -> $sub-start {
```

Comment:
Not sure how much choice we get about the JSON representation, but maybe it would be better to arrange it so the instance is explicitly given in the JSON rather than having to extract it?

Reply:
We can define it as we want.
The idea was, to minimize the amount of data to transfer.
What structure would you recommend?

Reply:
I wondered about something like:

* For single-instance modules, the top-level keys maps to a hash with the input values
* For multi-instance modules, the part of the taxonomy before the instance maps to an array, which carries an object per instance, with keys `name` (for the instance name) and values, which is then an object mapping the part of the taxonomy after the instance name to an object the the input values

So that we never repeat the instance name or the part of the taxonomy that comes before it.